src/core/chart.js

```diff
@@ -0,0 +1,252 @@
+  //and individual-timeseries-level display options.
+  var c3Data = {
+      columns: [],
+      types: {},
+      labels: {},
+      axes: {}
+  };
+
+  var yUnits = "";
+  var y2Units = "";
+  
+  var getTimeseriesName = shortestUniqueTimeseriesNamingFunction(metadata, data);
+  
+  //Add each timeseries to the graph
+  for(var i = 0; i < data.length; i++) {
+
+    //get metadata for this timeseries
+    var timeseries = data[i];
+    var timeseriesMetadata = _.find(metadata, function(m) {return m.unique_id === timeseries.id;});  
+    var timeseriesName = getTimeseriesName(timeseriesMetadata);
+       
+    //add the actual data to the graph
+    c3Data.columns.push([timeseriesName].concat(getMonthlyData(timeseries.data, timeseriesMetadata.timescale)));
+    
+    //monthly data is displayed as a line graph, but yearly and seasonal
+    //display as step graphs.
+    c3Data.types[timeseriesName] = timeseriesMetadata.timescale == "monthly" ? "line" : "step";
+
+    //Each timeseries needs to be associated with a y-axis.
+    //Two different variables measured with the same units, 
+    //like tasmin (degrees C) and tasmax (degrees C)
+    //can share a y-axis. 
+    //Variables with different units require seperate axes.
+    //C3 can theoretically support indefinite numbers of y-axes,
+    //but that would be hard for a user to make sense of, 
+    //so it's capped at two here.
+    if((!yUnits) || timeseries.units == yUnits) {
+      yUnits = timeseries.units;
+      c3Data.axes[timeseriesName] = "y";
+    }
+    else if((!y2Units) || timeseries.units == y2Units) {
+      y2Units = timeseries.units;
+      c3Data.axes[timeseriesName] = "y2";
+    }
+    else {
+      throw new Error("Error: too many data axes required for graph");
+    }
+  }
+  
+  //whole-graph display options: axis formatting and tooltip behaviour
+  var c3Axis = {};
+  c3Axis.x = monthlyXAxis;
+  c3Axis.y = formatYAxis(yUnits);
+  if(y2Units) { 
+    c3Axis.y2 = formatYAxis(y2Units);
+    }
+    
+  var c3Tooltip = {format: {}};
+  c3Tooltip.grouped = "true";
+  c3Tooltip.format.value = tooltipDisplayNumbersWithUnitsFunction(c3Data.axes, c3Axis);
+  
+  return {
+    data: c3Data,
+    tooltip: c3Tooltip,
+    axis: c3Axis
+  }; 
+};
+
+/*
+ * Helper function for parseTimeseriesForC3.
+ * Accepts a dataseries object with 1, 4, or 12 timestamp:value pairs
+ * and returns an array with twelve values in order by timestamp,
+ * repeating values as necessary to get a monthly-resolution sequence.
+ */
+var getMonthlyData = function(data, timescale = "monthly") {
+
+  var expectedTimestamps = {"monthly": 12, "seasonal": 4, "yearly": 1};
+  var monthlyData = [];
+  var timestamps = Object.keys(data).sort();
+  
+  if(timestamps.length == 17) {
+    throw new Error("Error: concatenated 17-point chronology.");
+  }
+  
+  if(timestamps.length != expectedTimestamps[timescale]) {
+    throw new Error("Error: inconsistent time resolution in data");
```

Comment:
Also, what catches these errors and what is done about it? Sorry if the answer ought to be obvious.